transformer_engine/pytorch/cpp_extensions/gemm.py

```diff
@@ -259,3 +259,159 @@ def gemm(
     _ = fn(*args)
 
     return out, grad_bias, gelu_input
+
+
+def grouped_gemm(
+    A: List[torch.Tensor],
+    B: List[torch.Tensor],
+    out: List[torch.Tensor],
+    dtype: torch.dtype,
+    workspaces: List[torch.Tensor],
+    gelu: bool = False,
+    gelu_input: Optional[List[torch.Tensor]] = None,
+    grad: bool = False,
+    accumulate: bool = False,
+    layout: str = "TN",
+    bias: Optional[List[torch.Tensor]] = None,
+    use_bias: bool = False,
+) -> Tuple[Union[List[torch.Tensor], None], ...]:
+    """Non FP8 Grouped GEMM."""
+
+    assert layout in ("TN", "NN", "NT"), f"GEMM layout {layout} not supported."
+    transa = layout[0] == "T"
+    transb = layout[1] == "T"
+    num_gemms = len(A)
+    empty_tensor = torch.Tensor()
+    empty_tensors = [torch.Tensor()] * num_gemms
+
+    if gelu and not grad:
+        gelu_input = [torch.empty_like(o, dtype=bias_dtype) for o in out]
+    elif not gelu:
+        gelu_input = empty_tensors
+
+    if grad and use_bias:
+        grad_bias = [
+            torch.empty(B[i].shape[1], dtype=out[0].dtype, device="cuda")
+            for i in range(num_gemms)
+        ]
+    else:
+        grad_bias = empty_tensors
+
+    bias = bias if use_bias else empty_tensors
+
+    assert (
+        A[0].dtype == dtype and B[0].dtype == dtype
+    ), f"Expected dtype={dtype}, but found A.dtype={A[0].dtype} and B.dtype={B[0].dtype}"
+    input_dtype = TE_DType[dtype]
+    output_dtype = TE_DType[out[0].dtype]
+    if use_bias:
+        bias_dtype = TE_DType[grad_bias[0].dtype] if grad else TE_DType[bias[0].dtype]
+    else:
+        bias_dtype = output_dtype
+
+    tex.te_grouped_gemm(
```

Comment:
Kindly suggest registering these functions (`te_grouped_gemm` and `te_fp8_grouped_gemm`) with TorchScript to enable the compute graph trace (See L232).

Reply:
Added.

Out of curiosity, is TorchScript registration in TE used for ONNX export, as I think TorchScript itself is depracated?

Reply:
Hi, the TS is for ONNX export, indeed.